server/metrics/prometheus.go

```diff
@@ -1,28 +1,96 @@
 package metrics
 
 import (
-	"github.com/jitsucom/jitsu/server/logging"
+	"os"
 	"strings"
+	"time"
+
+	"github.com/jitsucom/jitsu/server/logging"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/spf13/viper"
 )
 
-var Enabled = false
-
-func Init(enabled bool) {
-	Enabled = enabled
-	if Enabled {
-		logging.Info("✅ Initializing Prometheus metrics..")
-		initEvents()
-		initSourcesPool()
-		initSourceObjects()
-		initMetaRedis()
-		initCoordinationRedis()
-		initEventsRedis()
-		initUsersRecognitionQueue()
-		initUsersRecognitionRedis()
-		initStreamEventsQueue()
-	} else {
-		logging.Info("❌ Prometheus metrics reporting is not enabled. Read how to enable them: https://jitsu.com/docs/other-features/application-metrics")
+var Exported bool
+
+var Registry *prometheus.Registry
+
+func Enabled() bool {
+	return Registry != nil
+}
+
+func NewCounterVec(opts prometheus.CounterOpts, labels []string) *prometheus.CounterVec {
+	vec := prometheus.NewCounterVec(opts, labels)
+	Registry.MustRegister(vec)
+	return vec
+}
+
+func NewGaugeVec(opts prometheus.GaugeOpts, labels []string) *prometheus.GaugeVec {
+	vec := prometheus.NewGaugeVec(opts, labels)
+	Registry.MustRegister(vec)
+	return vec
+}
+
+const Unknown = "unknown"
+
+func Init(exported bool) {
+	logging.Info("✅ Initializing Prometheus metrics..")
+
+	Exported = exported
+	Registry = prometheus.NewRegistry()
+	Registry.MustRegister(
+		collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}),
+		collectors.NewGoCollector(),
+	)
+
+	initEvents()
+	initSourcesPool()
+	initSourceObjects()
+	initMetaRedis()
+	initCoordinationRedis()
+	initEventsRedis()
+	initUsersRecognitionQueue()
+	initUsersRecognitionRedis()
+	initStreamEventsQueue()
+}
+
+func InitRelay(clusterID string, viper *viper.Viper) *Relay {
+	if viper.GetBool("disabled") {
+		logging.Debugf("Metrics relay is disabled")
+		return nil
+	}
+
+	url := viper.GetString("url")
+	if url == "" {
+		logging.Debugf("Metrics relay URL is empty, disabling relay")
+		return nil
 	}
+
+	deploymentID := viper.GetString("deployment_id")
+	if deploymentID == "" {
+		deploymentID = clusterID
+	}
+
+	hostID, err := os.Hostname()
+	if err != nil {
+		logging.Debugf("Failed to get hostname for metrics relay, using '%s': %s", Unknown, err)
+		hostID = Unknown
+	}
+
+	timeout := time.Second
+	if viper.IsSet("timeout") {
+		timeout = viper.GetDuration("timeout")
+	}
+
+	relay := Relay{
+		URL:          url,
+		HostID:       hostID,
+		DeploymentID: deploymentID,
+		Timeout:      timeout,
+	}
+
+	logging.Debugf("✅ Initialized metrics relay: %+v", relay)
```

Comment:
Since it is an important configuration info - all logs here can have level `info`

Reply:
These logs are generated by service which should require the least amount of user attention. Since it is also enabled by default, moving the logs to INFO level would draw unwarranted attention (more so in case of problems the user actually has no way or desire to solve, which may also lead to disabling the relay altogether – and we sort of need it).